### PR TITLE
Implement recipe logic properly

### PR DESCRIPTION
**Pull request by HakkyuKim**
Follows the original semantics of recipe:
- plugins not in the recipe file are not recognized by the tool during integration test.
- plugins with empty list are marked 'excluded' during integration test.

### CODE REVIEW

File: tools/lib/src/recipe.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      }
+    }
+    return Recipe._(
+      profilesPerPackage,
+      excluded: excluded,
+    );
+  }
+
+  final List<String> _excluded;
+  final Map<String, List<Profile>> _profilesPerPackage;
```
Comment (swift-kim) [inline tools/lib/src/recipe.dart:39]: You may simplify this class by removing `_excluded` and allowing `_profilesPerPackage` to contain empty lists.

In that case `isExcluded` will be just
```dart
  bool isExcluded(String plugin) =>
      isRecognized(plugin) && _profilesPerPackage[plugin]!.isEmpty;
```

File: tools/test/recipe_test.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  b: ["wearable-5.5", "tv-6.0"]
+
+  c: []
+''';
+  });
+
+  test('correctly parses recipe', () {
+    final Recipe recipe = Recipe.fromYaml(loadYaml(yamlString) as YamlMap);
+
+    expect(recipe.isRecognized('c'), true);
```
Comment (swift-kim) [inline tools/test/recipe_test.dart:25]: You may also test for `'d'` (not present in the recipe).

**Review by swift-kim — COMMENTED** [review]
Just some nitpicks.

**Comment by HakkyuKim** [comment]
Analyze workflow will pass after #372.

**Review by swift-kim — APPROVED** [review]
Thank you!

**Pull request merged**